File: java/src/main/java/com/dddeurope/niuz/newsroom/Author.java
package com.dddeurope.niuz.newsroom;

import java.util.Objects;

public class Author {
    private final String id;
    private final String name;
    private final String bankAccount;
    private final String contractType;
    private final int rate;

    public Author(String id, String name, String bankAccount, String contractType, int rate) {
        this.id = id;
        this.name = name;
        this.bankAccount = bankAccount;
        this.contractType = contractType;
        this.rate = rate;
    }

    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getBankAccount() {
        return bankAccount;
    }

    public int getRate() {
        return rate;
    }

    public boolean paysBySubmission() {
        return "pay-by-submission".equals(contractType);
    }

    public boolean paysByPublication() {
        return "pay-by-publication".equals(contractType);
    }

    @Override
    public String toString() {
        return "Author{" +
                "id='" + id + '\'' +
                ", name='" + name + '\'' +
                ", bankAccount='" + bankAccount + '\'' +
                ", contractType='" + contractType + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Author author = (Author) o;
        return Objects.equals(getId(), author.getId());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getId());
    }
}

File: dotnet/niuz.application/newsroom/Author.cs
namespace niuz.application.newsroom
{
    public class Author
    {
        public string Id { get; }
        public string Name { get; }


        public Author(string id, string name)
        {
            Id = id;
            Name = name;
        }

        protected bool Equals(Author other)
        {
            return Id == other.Id;
        }

        public override bool Equals(object obj)
        {
            if (ReferenceEquals(null, obj)) return false;
            if (ReferenceEquals(this, obj)) return true;
            if (obj.GetType() != this.GetType()) return false;
            return Equals((Author) obj);
        }

        public override int GetHashCode()
        {
            return (Id != null ? Id.GetHashCode() : 0);
        }
    }
}